server/src/main/java/io/crate/expression/scalar/bitwise/BitwiseFunctions.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar.bitwise;
+
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.expression.scalar.arithmetic.BinaryScalar;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.functions.Signature;
+import io.crate.sql.tree.BitString;
+import io.crate.types.BitStringType;
+import io.crate.types.DataType;
+import io.crate.types.TypeSignature;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.function.BinaryOperator;
+
+import static io.crate.types.DataTypes.BYTE;
+import static io.crate.types.DataTypes.INTEGER;
+import static io.crate.types.DataTypes.LONG;
+import static io.crate.types.DataTypes.SHORT;
+
+public class BitwiseFunctions {
+
+    public record TypedOperator<T>(DataType<T> type, BinaryOperator<T> operator) {}
+
+    private enum Operations {
+        AND(
+            new TypedOperator<>(LONG, (a, b) -> a & b),
+            new TypedOperator<>(INTEGER, (a, b) -> a & b),
+            new TypedOperator<>(SHORT, (a, b) -> (short) (a & b)), // Bitwise operations on short and byte primitives are auto-casted to int, need to cast back.
+            new TypedOperator<>(BYTE, (a, b) -> (byte) (a & b)),
+            new TypedOperator<>(BitStringType.INSTANCE_ONE, (a, b) -> {
```

Comment:
Initially had a concern that we are registering with default length and might need to preserve BIT len of the bound signature on scalar registering, similar to `NumericSumAggregation` but I think it's not needed here.

If expression looks like `B'111'::BIT(2) & B'111'::BIT(2)`  it returns `B'11'` as expected as casts are done before bit operators.

I think one use case when we need to take bit length from the bound signature is when we call some function on bit column casted to a different length, say `bit_count(bit_col::bit(x))` - but this is not related to bitwise operations.

Also randomized bit length in dedicated test in https://github.com/crate/crate/pull/13278/files#diff-582123804cba3c099c1ec696fa472a63c99249c6718d9e2c7afeb65d4080b081R107